addon/mixins/gatekeeper.js

```diff
@@ -187,6 +185,21 @@ export default Mixin.create({
     return result;
   }),
 
+  /**
+   * Return the full url to the Portal's REST API
+   */
+  portalRestUrl: computed('isAuthenticated', function () {
+    let result;
+    if (this.get('isAuthenticated')) {
+      result = getPortalRestUrl(this.get('portal'));
+    } else {
+      const config = getOwner(this).resolveRegistration('config:environment');
+      result = config.torii.providers['arcgis-oauth-bearer'].portalUrl;
```

Comment:
this whole fetching of the config seems to be redundant b/c you just set it to `${this.get('portalHostname')}/sharing/rest` below. we should remove either one or the other. if we go w/ the latter, we should probably also add `'portalHostname'` to the CP's dependencies.

Reply:
good catch! currently this CP is not used, but it would have been wrong b/c `${this.get('portalHostname')}` does not have protocol.

Instead it *should* use the url from the config and simply wack `/sharing/rest` on the end